data/transactions/logic/TEAL_opcodes.md

```diff
@@ -1560,3 +1560,45 @@ For boxes that exceed 4,096 bytes, consider `box_create`, `box_extract`, and `bo
 | 0 | BlkSeed | []byte |  |
 | 1 | BlkTimestamp | uint64 |  |
 
+
+## ec_add v
+
+- Opcode: 0xe0 {uint8 curve}
+- Stack: ..., A: []byte, B: []byte &rarr; ..., []byte
+- for (curve points A and B) return the curve point A + B
+- **Cost**:  BN254=0 BN254_G1=10 BN254_G2=10 BLS12_381=0 BLS12_381_G1=20 BLS12_381_G2=20
+- Availability: v9
+
+`EC` Curves:
+
+| Index | Name | Notes |
+| - | ------ | --------- |
+| 0 | BN254 | BN254 curve, sometimes called BN256 |
+| 1 | BN254_G1 | G1 of the BN254 curve |
+| 2 | BN254_G2 | G2 of the BN254 curve |
+| 3 | BLS12_381 | BLS 12-381 curve |
+| 4 | BLS12_381_G1 | G1 of the BLS 12-381 curve |
+| 5 | BLS12_381_G2 | G2 of the BLS 12-381 curve |
+
+
+A, B are curve points in G1 group. Each point consists of (X, Y) where X and Y are 256 bit integers, big-endian encoded. The encoded point is 64 bytes from concatenation of 32 byte X and 32 byte Y.
+
+## ec_scalar_mul v
+
+- Opcode: 0xe1 {uint8 curve}
+- Stack: ..., A: []byte, B: []byte &rarr; ..., []byte
+- for (curve point A, scalar K) return the curve point KA
+- **Cost**:  BN254=0 BN254_G1=100 BN254_G2=100 BLS12_381=0 BLS12_381_G1=200 BLS12_381_G2=200
+- Availability: v9
+
+A is a curve point in G1 Group and encoded as described in `ec_add`. Scalar K is a big-endian encoded big integer that has no padding zeros.
```

Comment:
Same checks as for `ec_add`.

Do you want to bound the maximum size of `K`?
Why restricting `K` to have no padding zeros?

Reply:
I don't see a reason for the padding thing. I removed it.

I'll have to benchmark this for large K. I'm guessing the cost is log(K)? If restricted, what is a reasonable restriction, K maybe can be 32 or 64 bytes? (does it just need to be as big as the order of the group?)

Reply:
Commenting before resolving: The limit for scalars is 32 bytes.  This might have to become curve specific someday, if we have curves larger than that.